ui/src/containers/BeamlineActionsContainer.jsx

```diff
@@ -1,164 +1,136 @@
 import BeamlineActionDialog from '../components/BeamlineActions/BeamlineActionDialog';
 import { RUNNING } from '../constants';
 
-class BeamlineActionsContainer extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.plotIdByAction = {};
+function BeamlineActionsContainer(props) {
+  const { actionsList } = props;
 
-    this.startAction = this.startAction.bind(this);
-    this.hideOutput = this.hideOutput.bind(this);
-    this.newPlotDisplayed = this.newPlotDisplayed.bind(this);
-  }
+  const dispatch = useDispatch();
+  const currentAction = useSelector(
+    (state) => state.beamline.currentBeamlineAction,
+  );
 
-  startAction(cmdName, showOutput = true) {
-    const parameters = [];
+  const [plotIdByAction, setPlotIdByAction] = useState({});
 
-    this.props.actionsList.some((cmd) => {
-      if (cmd.name === cmdName) {
-        cmd.arguments.forEach((arg) => {
-          if (arg.type === 'float') {
-            parameters.push(Number.parseFloat(arg.value));
-          } else {
-            parameters.push(arg.value);
-          }
-        });
-        return true;
+  const startAction = useCallback(
+    (cmdName, showOutput = true) => {
+      const cmd = actionsList.find((cmd) => cmd.name === cmdName);
+      if (!cmd) {
+        return;
       }
-      return false;
-    });
-
-    this.plotIdByAction[this.props.currentAction.name] = null;
-    this.props.startAction(cmdName, parameters, showOutput);
-  }
 
-  hideOutput() {
-    this.props.hideOutput(this.props.currentAction.name);
+      const parameters = cmd.arguments.map((arg) =>
+        arg.type === 'float' ? Number.parseFloat(arg.value) : arg.value,
+      );
+
+      setPlotIdByAction((prev) => ({ ...prev, [currentAction.name]: null }));
+      dispatch(startBeamlineAction(cmdName, parameters, showOutput));
+    },
+    [actionsList, currentAction.name, dispatch],
+  );
+
+  const hideOutput = useCallback(() => {
+    dispatch(hideActionOutput(currentAction.name));
+  }, [currentAction.name, dispatch]);
+
+  const newPlotDisplayed = useCallback(
+    (plotId) => {
+      setPlotIdByAction((prev) => ({ ...prev, [currentAction.name]: plotId }));
+    },
+    [currentAction.name],
+  );
+
+  if (actionsList.length === 0) {
+    return null;
   }
 
-  newPlotDisplayed(plotId) {
-    this.plotIdByAction[this.props.currentAction.name] = plotId;
-  }
-
-  render() {
-    const currentActionRunning = this.props.currentAction.state === RUNNING;
-    const currentActionName = this.props.currentAction.name;
-
-    const defaultDialogPosition = { x: -100, y: 100 };
-
-    if (this.props.actionsList.length === 0) {
-      return null;
-    }
-
-    return (
-      <>
-        <Dropdown
-          title="Beamline Actions"
-          id="beamline-actions-dropdown"
+  const currentActionRunning = currentAction.state === RUNNING;
+  const currentActionName = currentAction.name;
+  const defaultDialogPosition = { x: -100, y: 100 };
```

Comment:
Can be extracted as a constant `DEFAULT_DIALOG_POSITION` at the top of the file.